lib/scala/kg/src/main/scala/data/kg/KgtkData.scala

```diff
@@ -0,0 +1,38 @@
+package data.kg
+
+import models.kg.{KgEdge, KgNode}
+import scala.collection.mutable.HashMap
+
+class KgtkData(edges: List[KgEdge], nodes: List[KgNode])
+  extends KgData(edges, nodes, List()) {
+
+  def this(data: Tuple3[List[KgEdge], List[KgNode], List[KgNode]]) =
+    this(edges = data._1, nodes = KgtkData.reduceNodes(data._2, data._3))
+
+  def this(resource: KgtkDataResource) =
+    this(data = resource.read())
+}
+
+object KgtkData {
+  def reduceNodes(nodes1: List[KgNode], nodes2: List[KgNode]): List[KgNode] = {
+    val nodesById = HashMap[String, KgNode]()
+    (nodes1 ::: nodes2).foreach(node => {
+      if (nodesById.contains(node.id))
+        nodesById(node.id) = mergeNodes(nodesById(node.id), node)
+      else
+        nodesById += (node.id -> node)
+    })
+    nodesById.values.toList
+  }
+
+  def mergeNodes(node1: KgNode, node2: KgNode) =
+    KgNode(
+      aliases = Some((node1.aliases getOrElse List()) ::: (node2.aliases getOrElse List()) distinct),
```

Comment:
Should be None if the list is empty. You can use Some(someList).filter(_.nonEmpty) to convert a list to a None if it's empty.

Reply:
Also, aliases should go away on a subsequent iteration. There should only be `labels`. We can synthesize a label field on the GraphQL side to pick the first one out.

Reply:
Removed aliases